app/views/upload.scala.html

```diff
@@ -43,15 +43,20 @@ <h2 class="govuk-error-summary__title logged-out-title" id="logged-out-error-tit
                 <div class="drag-and-drop">
                     <div class="govuk-summary-list govuk-file-upload">
                         <div class="govuk-summary-list__row">
-                            <dd id="drag-and-drop-success" class="govuk-summary-list__value drag-and-drop__success" hidden tabindex="-1" role="alert" aria-describedby="successMessageText">
+                            <dd id="drag-and-drop-success" class="govuk-summary-list__value drag-and-drop__success" hidden
+                                tabindex="-1" role="alert" aria-describedby="success-message-text">
                                 <div>
                                     @greenTickMark()
-                                    <p id="successMessageText" >@Html(Messages("upload.dragAndDropSuccessMessage"))</p>
+                                    <p id="success-message-text" >@Html(Messages("upload.dragAndDropSuccessMessage"))</p>
                                 </div>
                             </dd>
-                            <dd id="drag-and-drop-failure" class="govuk-summary-list__value drag-and-drop__failure" hidden tabindex="-1" role="alert" aria-describedby="failureMessageText">
-                                @redWarningSign()
-                              <p id="failureMessageText">@Messages("upload.dragAndDropErrorMessage")</p>
+                            <dd id="drag-and-drop-failure" class="govuk-summary-list__value drag-and-drop__failure" hidden
+                                tabindex="-1" role="alert" aria-describedby="failure-message-text">
+                                <div id="failure-message-text">
+                                    @redWarningSign()
+                                    <p id="non-folder-selected-message-text" hidden>@Messages("upload.dragAndDropErrorMessage")</p>
+                                    <p id="submission-without-a-folder-message-text" hidden>@Messages("upload.submissionWithoutAFolderMessageText")</p>
```

Comment:
This is a little bit tricky to follow, because the JavaScript has to show and hide the failure container and the individual messages. It's also slightly confusing because the `submission-without-a-folder-message-text` element isn't solely related to the drag-and-drop, but it's nested in an element that sounds drag-and-drop specific.

What about having two separate `<dd>` elements so that you can just show and hide those individually? They'll be very similar, but you could extract a sub-view to commonise the code.

Reply:
This is a really good idea! Thanks.